fuel/datasets/hdf5.py

```diff
@@ -106,16 +106,20 @@ class H5PYDataset(Dataset):
          array
       4. ``stop`` : stop index (exclusive) of the split in the source
          array
-      5. ``available`` : boolean, ``False`` is this split is not available
+      6. ``indices`` : h5py.Reference, reference to a dataset containing
+         subset indices for this split/source pair.
+      6. ``available`` : boolean, ``False`` is this split is not available
          for this source
-      6. ``comment`` : comment string
+      7. ``comment`` : comment string
 
     Parameters
     ----------
     file_or_path : :class:`h5py.File` or str
         HDF5 file handle, or path to the HDF5 file.
-    which_set : str
-        Which split to use.
+    which_set : str or tuple of str
+        Which split(s) to use. If one than more split is requested,
+        the provided sources will be the intersection of provided
+        sources for these splits.
     subset : slice, optional
```

Comment:
This is not related to this PR, but why is this `subset` option required? I thought such things should be done at the datastream level.